myTest.java

```diff
@@ -0,0 +1,177 @@
+    public void setOriginalSystemOut(){
+        System.setOut(originalOut);
+    }
+
+    @BeforeEach
+    public void setUP()  {
+        ilyaCart = new Cart("ilya-cart");
+
+        car = new RealItem();
+        car.setName("KIA");
+        car.setPrice(20000.0);
+        car.setWeight(1600.0);
+        realItemCar  = "Class: class shop.RealItem; Name: KIA; Price: 20000.0; Weight: 1600.0";
+
+        game = new VirtualItem();
+        game.setName("Star Wars");
+        game.setPrice(100);
+        game.setSizeOnDisk(36000.0);
+        virtualItemGame = "Class: class shop.VirtualItem; Name: Star Wars; Price: 100.0; Size on disk: 36000.0";
+
+        itemsAll = realItemCar + "\r\n" + virtualItemGame + "\r\n";
+
+        ilyaCart.addRealItem(car);
+        ilyaCart.addVirtualItem(game);
+
+        expectedPrice = game.getPrice() + game.getPrice()*TAX + car.getPrice() + car.getPrice()*TAX;
+
+        ilyaCartObjectContent = gson.toJson(ilyaCart);
+
+        System.setOut(new PrintStream(outShowItemsContent));
+    }
+
+    @Test
+    @Order(3)
+    @Tag("Cart")
+    public void testCartName() {
+        assertEquals("ilya-cart", ilyaCart.getCartName());
+    }
+
+    @Test
+    @Order(4)
+    @Tag("Cart")
+    public void testTotalPrice() {
+        assertEquals(expectedPrice, ilyaCart.getTotalPrice());
+    }
+
+    @Test
+    @Order(5)
+    @Tag("Cart")
+    public void testShowItems() {
+        ilyaCart.showItems();
+        assertEquals(itemsAll, outShowItemsContent.toString());
+    }
+
+    @Test
+    @Order(6)
+    @Tag("Cart")
+    public void testDeleteRealItem() {
+        ilyaCart.deleteRealItem(car);
+        ilyaCart.showItems();
+        assertEquals(virtualItemGame+"\r\n", outShowItemsContent.toString());
+    }
+
+    @Test
+    @Order(7)
+    @Tag("Cart")
+    public void testDeleteVirtualItem() {
+        ilyaCart.deleteVirtualItem(game);
+        ilyaCart.showItems();
+        assertEquals(realItemCar+"\r\n", outShowItemsContent.toString());
+    }
+
+    @Test
+    @Tag("RealItem")
+    @Order(1)
+    public void testRealItem(){
+        assertAll("Should return name, weight and price of the car",
+                ()-> assertEquals("KIA",car.getName()),
+                ()-> assertEquals(20000.0, car.getPrice()),
+                ()-> assertEquals(1600.0, car.getWeight())                    ,
+                ()-> assertEquals(realItemCar , car.toString())
+        );
+    }
+
+    @Test
+    @Tag("VirtualItem")
+    @Order(2)
+    public void testVirtualItem(){
+        assertAll("Should return name, weight and price of the game",
+                ()-> assertEquals("Star Wars", game.getName()),
+                ()-> assertEquals(100.0, game.getPrice()),
+                ()-> assertEquals(36000.0, game.getSizeOnDisk()),
+                ()-> assertEquals(virtualItemGame, game.toString())
+        );
+    }
+
+    @Disabled
+    @Order(10)
+    @Tag("Parser")
+    @Test
+    public void noSuchFileExceptionTest(){

```

Comment:
Now make this test parameterized with at least 5 different datasets (let me know if you have difficulties with this comment)

Reply:
I've made the parameterized test, but I'm still confusing which datasets will be necessary for this particular test.
I've read JUnit 5 User Guide several times(especially 2.15.*), but still didn't found a solution.
![image](https://user-images.githubusercontent.com/63204236/79322984-47de9900-7f16-11ea-822a-0844c09a831a.png)

Reply:
For example: unix path to files, incorrect windows path, path to file without extension and so on. Please, reach me in skype, if you need more details

Reply:
More than 5 datasets were added.